### PR TITLE
Add Nebari

### PR DESCRIPTION
**Pull request by viniciusdc**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by viniciusdc** [comment]
@conda-forge-admin, please restart ci

File: recipes/nebari/meta.yaml
Comment (viniciusdc) [inline recipes/nebari/meta.yaml:51]:
```suggestion
    - nebari --version
```

File: recipes/nebari/meta.yaml
Comment (viniciusdc) [inline recipes/nebari/meta.yaml:15]:
```suggestion
  noarch: python
  skip: true # [win]
```

**Comment by viniciusdc** [comment]
@jaimergp could you have a look at this? I would like to know if this follows what was expected of the new noarch customizations builds... I think I did something wrong there

**Comment by iameskild** [comment]
I agree to be a maintainer of this project 👍 

cc @viniciusdc

File: recipes/nebari/meta.yaml
Comment (ndevenish) [inline recipes/nebari/meta.yaml:61]:
```suggestion
  license: BSD-3-Clause
  license_family: BSD
```

**Comment by ndevenish** [comment]
The nebari package on pypi includes the `LICENSE` file, so you don't need to include it in this feedstock (conda-build will pick it up from the package)

File: recipes/nebari/meta.yaml
Comment (ndevenish) [inline recipes/nebari/meta.yaml:26]: I haven't seen this noarch-package restriction used before. What does this do on it's own? The page you linked to seems to suggest that it must have a selector e.g.
```
- __unix # [unix]
```
and that it is for using when you have an architecture-specific dependency. If it's just that one of the dependencies isn't available on windows - the old way was to just leave it in and so if that dependency ever did become available on windows, then it'd automatically qualify (since noarch is by default built on linux only, it would be fine).

If it explicitly doesn't work on windows, then maybe this is the approach to use...

Comment (BastianZim) [reply recipes/nebari/meta.yaml:26]: It's from here: https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-virtual.html
Basically, it allows the package only to be installed in an environment where this is available and can be used if a package only works on Linux, for example. Keeps it noarch and reduces our resources.

I've been using this for at least a couple of months in a couple of packages, and it works really well! 

One example: https://github.com/conda-forge/smac-feedstock/blob/048bfd759b09f78f0d60f3c952e49261b48bf629/recipe/meta.yaml#L37

File: recipes/nebari/meta.yaml
Comment (BastianZim) [inline recipes/nebari/meta.yaml:53]: Nice! I'm wondering if it wouldn't make sense to include that in Grayskull, because `setuptools_scm`etc. is sometimes not included which this would catch.

File: recipes/nebari/meta.yaml
Comment (goanpeca) [inline recipes/nebari/meta.yaml:24]:
```suggestion
    - setuptools-scm
```

File: recipes/nebari/meta.yaml
Comment (dopplershift) [reply recipes/nebari/meta.yaml:26]: Yes, my understanding is that you don't need (or even want for noarch) the selector.

Comment (BastianZim) [reply recipes/nebari/meta.yaml:26]: Yes, the selector shouldn't be used. Otherwise, the selector will remove the virtual package on windows, making it available again on all platforms.

Comment (viniciusdc) [reply recipes/nebari/meta.yaml:26]: Hi @BastianZim @dopplershift @ndevenish, thanks for the comments. Though I ended up being lost in conclusion... Regarding the use of `__unix`, based on the linked page and @BastianZim  comment above, it should work as a virtual dependency and restrict the package from being built on win, right?. Wich will not affect the package structure (maintaining its `noarch`)

On the other hand, using the `skip: True  # [win]` selectors are not advised as they are incompatible with `noarch`, and this is package is noarch. The only exception is that it does not support windows (it can be installed but can't be used...) 

..so I ended up losing track of which "selector" you all were referring to in this thread 😅

File: recipes/nebari/meta.yaml
Comment (viniciusdc) [reply recipes/nebari/meta.yaml:53]: Good idea, that could be an extra custom setting that the Grayskull config could load -- e.g custom test class

File: recipes/nebari/meta.yaml
Comment (BastianZim) [reply recipes/nebari/meta.yaml:26]: We really should add this to the docs, it is indeed becoming slightly confusing... 😄 
I'm just writing this down here once so that I can link later to it in case I need but maybe it's also helpful here.

There are two cases that are relevant here in which `arch` packages can be converted into `noarch` packages, both rely on virtual packages.

The first one is if there are different dependencies for different operating systems. In that case, https://github.com/conda-forge/conda-forge.github.io/pull/1839 can be used.

Let's assume that we have a package X that has a dependency Y but only on windows. In this example, OSX doesn't exist so I won't use Unix.

First, we instruct the CI to build the package on multiple different platforms, Linux and windows in this case (Through conda-forge.yml).
Then, we add the dependency using the `# [win]` [preprocessing](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#preprocessing-selectors) selector. This makes sure that the dependency is only added when we are building on windows. 

```yml
requirements:
  run:
    - Y  # [win]
```

Now, we have two packages, one built on Linux with no extra dependency and one built on Windows with the extra dependency.
Finally, we need to make sure that a Linux user cannot accidentally install the package that has the extra windows dependency. 
For that, we include two [virtual packages](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-virtual.html): `__linux` and `__win` in the run requirements. 
Virtual packages are only available on the platform they specify. Meaning if a package has a dependency on `__win`, the solver will check during the installation of the package if the solver is currently working on a windows system and only allows the installation to proceed if it really is on windows.
Since we can't have both in the same package, because then it couldn't be installed at all, we also add preprocessing selectors to both of them:
```yml
- __linux. # [linux]
- __win  # [win]
```
This, way we will have two packages. The first package is built on Linux and therefore has the `__linux` package and nothing else. The second one is built on Windows and therefore has the `__win` virtual package and the extra dependency.
When the end-user now installs the package, the solver will check that it gets the package that satisfies the virtual package, meaning the windows version on windows and the Linux version on Linux.
_Note:_ There are other virtual packages as well (Unix, OSX) but the approach is the same.

The meta.yaml is, therefore:

```yml
requirements:
  run:
    - Y  # [win]
    - __linux # [linux]
    - __win  # [win]
```
---
The second case is if we have a package that can only be used on a certain OS. This is commonly the case because it relies on system features that are not available on all OSs.

So far, this was done by using `skip: True` and the respective preprocessing selector: `# [win]`. That would mean that the package is only available on Linux and OSX.

Using the fact that virtual packages only allow the installation of a package if they exist, we can change this to a `noarch` solution through two steps.

First, the skip part needs to be removed and `noarch: python` added so that it is built as a noarch package.
Then, the virtual package is added to the run requirements.
This way, the package can only be installed on the OS that the virtual package allows. Here, the preprocessing selector shouldn't be used, otherwise, the virtual package will be removed again if it is built on a different platform, which we are trying to prevent.

```yml
requirements:
  run:
    - __linux
```

File: recipes/nebari/meta.yaml
Comment (BastianZim) [reply recipes/nebari/meta.yaml:53]: True, that works as well, but I just meant that Grayskull should add that to the recipes it generates so that the CI can catch this. :)

File: recipes/nebari/meta.yaml
Comment (viniciusdc) [reply recipes/nebari/meta.yaml:26]: That's wonderful docs here!! Thanks, @BastianZim. It could be added to the open docs @jaimergp created as supplementary information or a TLDR. What do you all think?

**Comment by viniciusdc** [comment]
@conda-forge/core, this is ready for review. Please keep the above conversation in mind during the review

File: recipes/nebari/meta.yaml
Comment (BastianZim) [reply recipes/nebari/meta.yaml:26]: My pleasure! :) Either one is fine for me. I can also wait until the other PR is merged and then add the info in a new PR.

**Comment by iameskild** [comment]
@conda-forge/core this is ready for a final review. Thank you :)

cc @viniciusdc

**Comment by BastianZim** [comment]
Do you need core approval? Otherwise, I'm happy to merge.

**Comment by iameskild** [comment]
@BastianZim @viniciusdc since you are both more familiar with the processes and workings of conda-forge, I'll let you decide we should proceed :)

**Comment by viniciusdc** [comment]
> Do you need core approval? Otherwise, I'm happy to merge.

Hi @BastianZim, if you have merge rights I would really appreciate if you can merge :smile:. thanks in advance. The CI tests for Linux are passing (wich is the target) and there are no problems with the recipe, thus I would say this is ready to go

File: recipes/nebari/LICENSE
Comment (BastianZim) [inline recipes/nebari/LICENSE:1]: This can be removed, it's already included in the source dist.

**Review by BastianZim — CHANGES_REQUESTED** [review]
One small thing and I'll merge. :)

File: recipes/nebari/LICENSE
Comment (viniciusdc) [reply recipes/nebari/LICENSE]: Hi @BastianZim, thanks, I just removed the License file.

**Pull request merged**
